pkg/scheduler/preemption/preemption_test.go

```diff
@@ -2796,3 +2813,83 @@ func TestPreemptionMessage(t *testing.T) {
 		}
 	}
 }
+
+func hierarchyManagerComparer(x, y hierarchy.Manager[*cache.ClusterQueueSnapshot, *cache.CohortSnapshot]) bool {
+	if !clusterQueuesSnapshotsComparer(x.ClusterQueues(), y.ClusterQueues()) {
+		return false
+	}
+	if !cohortSnapshotsComparer(x.Cohorts(), y.Cohorts()) {
+		return false
+	}
+	return true
+}
+
+func clusterQueuesSnapshotsComparer(x, y map[kueue.ClusterQueueReference]*cache.ClusterQueueSnapshot) bool {
+	if len(x) != len(y) {
+		return false
+	}
+	for name, xSnapshot := range x {
+		ySnapshot, ok := y[name]
+		if !ok {
+			return false
+		}
+		if !clusterQueueSnapshotComparer(xSnapshot, ySnapshot) {
+			return false
+		}
+	}
+	return true
+}
+
+func clusterQueueSnapshotComparer(x, y *cache.ClusterQueueSnapshot) bool {
+	// need to evaluate at every level of the hierarchy
+	opt := append(SnapCmpOpts(),
+		cmpopts.IgnoreFields(cache.ClusterQueueSnapshot{}, "ClusterQueue", "tasOnly"),
+		cmpopts.IgnoreFields(hierarchy.ClusterQueue[*cache.CohortSnapshot]{}, "cohort"),
+	)
+
+	diff := cmp.Diff(x, y, opt...)
+	if diff != "" {
```

Comment:
```suggestion
	if !cmp.Equal(x, y, opt...) {
```

You used cmp.Equal() in your custom Comparer. It is good point. I forgot about it. You right, it's much better to use it instead of cmp.Diff().